pkg/user/user.go

```diff
@@ -1,19 +1,29 @@
 package user
 
 import (
+	"errors"
+
+	"github.com/matthewhartstonge/argon2"
```

Comment:
- `golang.org/x/crypto/argon2` is better than `github.com/matthewhartstonge/argon2` because it's strongly recommended to use the official implementation for security.
- Use `golang.org/x/crypto/bcrypt`: I had said we would use Argon2, but I'm afraid I have to retract that: Argon2 requires various parameters to be encoded and they have to be stored with the hash. Bcrypt, on the other hand, requires no special parameters to encode and is very easy to use.